store/items.ts

```diff
@@ -0,0 +1,71 @@
+import Vue from 'vue';
+import { BaseItemDto } from '@jellyfin/client-axios';
+import { ActionTree, GetterTree, MutationTree } from 'vuex';
+import map from 'lodash/map';
+import some from 'lodash/some';
+import forEach from 'lodash/forEach';
+
+export interface ItemsState {
+  byId: Record<string, BaseItemDto>;
+  allIds: string[];
+}
+
+export const defaultState = (): ItemsState => ({
+  byId: {},
+  allIds: []
+});
+
+export const state = defaultState;
+
+export const getters: GetterTree<ItemsState, ItemsState> = {
+  getItem: (state) => (id: string): BaseItemDto | undefined => state.byId[id],
+  getItems: (state) => (ids: string[]): BaseItemDto[] =>
+    map(ids, (id) => state.byId[id])
+};
+
+export const mutations: MutationTree<ItemsState> = {
+  ADD_ITEM(state: ItemsState, { item }: { item: BaseItemDto }) {
+    if (!item.Id) {
+      throw new Error('No item ID');
+    }
+    Vue.set(state.byId, item.Id, item);
```

Comment:
Why we need this? We shouldn't need it in Vuex as everything is reactive and we're doing a single level deep.

https://forum.vuejs.org/t/vuex-best-practices-for-complex-objects/10143/2

Reply:
Cause it creates a key, and Vuex doesn't react to key creation. See point 2 in screenshot https://vuex.vuejs.org/guide/mutations.html 
![print_screen_2021-03-04-23-54-59](https://user-images.githubusercontent.com/1619359/110041842-13014d80-7d45-11eb-806a-32e572acb4b6.png)